### PR TITLE
Enroll Management Service: Server side validation

### PR DESCRIPTION
**Pull request by navya-BTC**

### CODE REVIEW

File: participant-datastore/enroll-mgmt-module/enroll-mgmt/src/main/java/com/google/cloud/healthcare/fdamystudies/controller/EnrollmentTokenController.java [hunk 1]
```diff
@@ -61,76 +62,61 @@
...
             ErrorResponseUtil.ErrorCodes.INVALID_INPUT.getValue(),
-            ErrorResponseUtil.ErrorCodes.TOKEN_REQUIRED.getValue(),
+            ErrorResponseUtil.ErrorCodes.UNKNOWN_TOKEN.getValue(),
             response);
+
+        enrollAuditEventHelper.logEvent(ENROLLMENT_TOKEN_FOUND_INVALID, auditRequest);
         return null;
       }
-    } else {
+    }
```
[the remark this answers is not in the document]
Comment (navya-BTC) [reply participant-datastore/enroll-mgmt-module/enroll-mgmt/src/main/java/com/google/cloud/healthcare/fdamystudies/controller/EnrollmentTokenController.java]: Fixed the comment.

File: participant-datastore/enroll-mgmt-module/enroll-mgmt/src/main/java/com/google/cloud/healthcare/fdamystudies/service/StudyStateServiceImpl.java [hunk 2]
```diff
@@ -208,43 +208,41 @@ public StudyStateRespBean saveParticipantStudies(
...
-                MyStudiesUserRegUtil.getIsoDateFormat(participantStudiesBO.getEnrolledDate()));
-          }
-          if (participantStudiesBO.getSite() != null) {
-            studyStateBean.setSiteId(participantStudiesBO.getSite().getId().toString());
-          }
-          serviceResponseList.add(studyStateBean);
+    List<ParticipantStudyEntity> participantStudiesList =
+        participantStudiesInfoDao.getParticipantStudiesInfo(userDetailsEntity.getUserId());
+    if (participantStudiesList != null && !participantStudiesList.isEmpty()) {
+      for (ParticipantStudyEntity participantStudiesBO : participantStudiesList) {
```
[the remark this answers is not in the document]
Comment (navya-BTC) [reply participant-datastore/enroll-mgmt-module/enroll-mgmt/src/main/java/com/google/cloud/healthcare/fdamystudies/service/StudyStateServiceImpl.java]: Fixed the comment.

**Comment by moschetti** [comment]
review conflicts in StudyStateController.java

**Comment by navya-BTC** [comment]
Resolved the conflict

**Review by moschetti — APPROVED** [review]

**Pull request merged**